backend/src/revisions/revisions.service.ts

```diff
@@ -219,4 +220,39 @@ export class RevisionsService {
       await this.revisionRepository.save(revision);
     }
   }
+
+  // Delete all old revisions everyday on 0:00 AM
+  @Cron('0 0 * * *')
+  async handleCron(): Promise<void> {
+    return await this.removeOldRevisions();
+  }
+
+  // Delete all old revisions 5 sec after startup
+  @Timeout(5000)
+  async handleTimeout(): Promise<void> {
+    return await this.removeOldRevisions();
+  }
+
+  async removeOldRevisions(): Promise<void> {
+    const currentTime = new Date().getTime();
+    const revisionRetentionDays: number = parseInt(process.env.HD_REVISION_RETENTION_DAYS || '0')
```

Comment:
I know this is still a WIP pull request and probably subject to change. However in case you haven't seen it, the environment variable processing and validation is better done via the config module.

Reply:
> via the config module.

Thank you for teaching me.
This is very helpful as I didn't realize that.